action-server/src/listeners/dbListeners.ts

```diff
@@ -0,0 +1,163 @@
+import { readFile } from "node:fs/promises";
+import type http from "node:http";
+import * as path from "node:path";
+import type { Pool, PoolClient } from "pg";
+import { configuration } from "../config";
+import { ActionsDbManager } from "../db";
+import { ActionWorkerPool } from "../threads/workerPool";
+import type { ActionDefinitionInsertedPayload, ActionResponse, ActionRunInsertedPayload } from "../type/types";
+import { extractSchemas } from "../utils/codeRunner";
+
+let listenClient: PoolClient | undefined;
+
+async function readFileFromStore(fileName: string): Promise<string> {
+  // read file from aerie file store and return [resolve] it as a string
+  const fileStoreBasePath = configuration().ACTION_LOCAL_STORE;
+  const filePath = path.join(fileStoreBasePath, fileName);
+  console.log(`path is ${filePath}`);
+  return await readFile(filePath, "utf-8");
+}
+
+async function handleActionDefinition(payload: ActionDefinitionInsertedPayload) {
+  console.log("action definition inserted");
+
+  // read the action file and extract parameter/setting schemas
+  const actionJS = await readFileFromStore(payload.action_file_path);
+  // console.debug(actionJS);
+  const schemas = await extractSchemas(actionJS);
+
+  console.info(`schemas ${JSON.stringify(schemas, null, 2)}`);
+
+  // todo: set schemas on the DB row?
+  const pool = ActionsDbManager.getDb();
+  const query = `
+    UPDATE actions.action_definition
+    SET
+      parameter_schema = parameter_schema || $1::jsonb,
+      settings_schema = settings_schema || $2::jsonb
+    WHERE id = $3
+      RETURNING *;
+  `;
+
+  try {
+    const res = await pool.query(query, [
+      JSON.stringify(schemas.paramDefs),
+      JSON.stringify(schemas.settingDefs),
+      payload.action_definition_id,
+    ]);
+    console.log("Updated action_definition:", res.rows[0]);
+  } catch (error) {
+    console.error("Error updating row:", error);
+  }
+}
+
+async function handleActionRun(payload: ActionRunInsertedPayload) {
```

Comment:
Nonblocking: Suggestion:

```suggestion
async function runAction(payload: ActionRunInsertedPayload) {
```

Reply:
Thank you for the suggestion. I renamed it